examples/controlnet/train_controlnet_flax.py

```diff
@@ -907,10 +907,15 @@ def compute_loss(params, minibatch, sample_rng):
 
             if args.snr_gamma is not None:
                 snr = jnp.array(compute_snr(timesteps))
-                if noise_scheduler.config.prediction_type == "v_prediction":
-                    # Velocity objective requires that we add one to SNR values before we divide by them.
-                    snr = snr + 1
-                snr_loss_weights = jnp.where(snr < args.snr_gamma, snr, jnp.ones_like(snr) * args.snr_gamma) / snr
+                if noise_scheduler.config.prediction_type == "epsilon":
+                    snr_loss_weights = jnp.where(snr < args.snr_gamma, snr, jnp.ones_like(snr) * args.snr_gamma) / snr
```

Comment:
Why do we have to compute `snr_loss_weights` two times?

Reply:
Well, I did not get the point. When `snr_gamma` is not specified, there is no need to compute `snr_loss_weights`.

Reply:
I am saying we always compute `snr_loss_weights` when `args.snr_gamma` is not None. Then we reuse it and make changes to it depending on the `noise_scheduler.config.prediction_type`.

Reply:
Get it. Do you mean exactly that we first create the weights as the default value min(SNR, gamma), and then we can divide it by SNR or SNR+1 respectively when we use eps/v-prediction?
I would say that both ways do not make much difference. Both are good to me.

Reply:
More concretely, I mean this:

```python
if args.snr_gamma is not None:
   snr = compute_snr(noise_scheduler, timesteps)
   mse_loss_weights = torch.stack([snr, args.snr_gamma * torch.ones_like(timesteps)], dim=1).min(dim=1)[0]
   
   if noise_scheduler.config.prediction_type == "v_prediction":
        mse_loss_weights = mse_loss_weights / (snr + 1)
```

Does that make sense?

Reply:
Of course. That's what I think exactly.